src/Microsoft.AspNetCore.Authentication.Abstractions/AuthenticateResult.cs

```diff
@@ -73,17 +78,35 @@ public static AuthenticateResult NoResult()
         /// <returns>The result.</returns>
         public static AuthenticateResult Fail(Exception failure)
         {
-            return new AuthenticateResult() { Failure = failure };
+            return new AuthenticateResult() { Error = new AuthenticationError(failure) };
         }
 
         /// <summary>
         /// Indicates that there was a failure during authentication.
         /// </summary>
-        /// <param name="failureMessage">The failure message.</param>
+        /// <param name="failure">The failure exception.</param>
+        /// <param name="properties">Additional state values for the authentication session.</param>
         /// <returns>The result.</returns>
-        public static AuthenticateResult Fail(string failureMessage)
+        public static AuthenticateResult Fail(Exception failure, AuthenticationProperties properties)
         {
-            return new AuthenticateResult() { Failure = new Exception(failureMessage) };
+            return new AuthenticateResult() { Error = new AuthenticationError(failure, properties) };
```

Comment:
Basically instead of adding an Error, just add a new _errorProperties field to use here, and return that instead of _Error.Properties

Reply:
They are the same properties. We just don't have a way to surface them for failure scenarios.

Reply:
A field can still work, just copy the properties in both Success and Failure instead of calling through to the Ticket/Error.

My main pushback is AuthenticateResult is already meant to handle errors.  We don't need another AuthenticationError class